webapi/src/main/java/org/sakaiproject/webapi/controllers/GradesController.java

```diff
@@ -95,7 +97,10 @@
                         double total = 0;
                         for (GradeDefinition gd : grades) {
                             if (gd.getGradeEntryType() == GradebookService.GRADE_TYPE_POINTS) {
-                                total += Double.parseDouble(gd.getGrade());
+                                String grade = gd.getGrade();
+                                if (!StringUtils.isBlank(grade)) {
+                                    total += Double.parseDouble(grade);
+                                }
```

Comment:
what's going on here? you are adding up total points?

Reply:
Yep. It's part of working out the average score for an assignment.

Reply:
Is precision important? i don't want to cause lots of work, but what about adding a method to edu-services to get average score for an item ?

Reply:
sorry i know intent of this jira was simple !

Reply:
Would the aim of doing that be to improve performance, doing it in edu services? Do you see that being useful elsewhere? I'll do it if so. When we make the dashboards the default landing page for users, that api call will be getting hit every time.

Reply:
i think my main intent is to make sure calculations like this only happen in exactly one place ...... i can definitely imagine a need for item averages  ...... i hate seeing stuff like this in multiple places with different rules around rounding (HALF_UP vs HALF_DOWN) and different rules around precision (e.g., doubles via BigDec)

Reply:
I reckon we should let this in though and maybe raise another JIRA about that. I'd want to update the average each time a grade was submitted and have it sat in a dedicated gb column, not calculated on the fly each time a user hits their dashboard. I'm presuming that's what you have in mind. A chunk of work.

Reply:
So, we're already doing it in the gradescontroller, so let's get this in and jira moving it to edu-services. I'm definitely interested in doing that, gathering it in one place. I didn't think at the time that it would be a widely wanted thing. It all seemed a bit ad-hoc as it was specced up.